### PR TITLE
chore(docs): reformat configuration page

### LINKED ISSUE #4396: Configuration documentation table hard to read
### Summary of problem
The documentation's [configuration page](https://ddtrace.readthedocs.io/en/stable/configuration.html) contains a table of all the possible variables that can be set. This table includes the default value. There are a few variables that default to long regular expressions which don't render with any extra line breaks. This causes the description column to be far to the right after a long scroll.

### Which version of dd-trace-py are you using?
n/a

### Which version of pip are you using?
n/a

### Which libraries and their versions are you using?
n/a

### How can we reproduce your problem?
Go to the [documentation website](https://ddtrace.readthedocs.io/en/stable/configuration.html)

### What is the result that you get?
A table that requires significant horizontal scrolling.

### What is the result that you expected?
Have the page render in a way that doesn't require significant horizontal scrolling. Alternately, move the default column to be the most right column. This would only require scrolling when people care about the few values that are long.

### PR DESCRIPTION
**Pull request by brettlangdon**
## Description
This change introduces a helper directive `.. ddtrace-configuration-options::` that will provide a consistent formatting from a list of configuration options. The reason we went with custom directive approach is the formatting of the configuration options can be changed at any time without needing to reformat each option individually.

## Motivation

The large table didn't handle large descriptions or default values very well. Especially if the values were code blocks since they do not overflow.

## Design 

The input to the directive is a YAML format of:

```yaml
OPTION_NAME:
    type: <type>
    default: <default>
    description: <description>
    version_added:
        vX.Y.Z:
```

If `type`is not provided then `String` is used.

If `default` is not provided then `(no value)` is used.

`description` **must** be provided.

`version_added` is a an optional mapping of versions to a message about what was changed. If no message is added then that is used as the version the configuration option was added, if there is a message then it it will render as a change.

Example usage of `version_added`:

```yaml
ENV_NAME:
    description: An env variable to use to set an option
    version_added:
        v1.2.3:
        v1.3.4: The value now can now contain ampersand.
        v2.0.0: |
            The option has been renamed from ``MY_ENV_NAME`` to ``ENV_NAME``.
```

`v1.2.3` is when the option was added for the first time, and it was updated in `v1.3.4`, and `v2.0.0`.

If the value of `default`, `description`, or `version_added` contains any special characters then a multi-line YAML block must be used.

```yaml
description: |
    ``inline code block``
```

The values of `type`, `default`, `version_added`, and `description` may contain rST formatted text.

## Relevant issue(s)

Fixes #4396 

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [x] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

**Comment by brettlangdon** [comment]
https://output.circle-artifacts.com/output/job/2c8d6310-2e04-4456-adb8-50590c31b640/artifacts/0/tmp/docs/configuration.html

File: docs/configuration.rst [hunk 1]
```diff
@@ -7,380 +7,343 @@
...
+         :ref:`falcon`. Added in ``v0.36.0``. See `Unified Service Tagging`_ for more information.
+
+   DD_SERVICE_MAPPING:
+     description: |
+         Define service name mappings to allow renaming services in traces, e.g. ``postgres:postgresql,defaultdb:postgresql``.
+
+   DD_TAGS:
+     description: |
+         Set global tags to be attached to every span. Value must be either comma or space separated. e.g. ``key1:value1,key2,value2`` or ``key1:value key2:value2``.
+         Comma separated support added in ``v0.38.0`` and space separated support added in ``v0.48.0``.
```
Comment (Kyle-Verhoog) [inline docs/configuration.rst:31]: could we add `version_added:` as a key in these? that would be awesome for helping remind us to add that valuable field.

**Review by Kyle-Verhoog — APPROVED** [review]
awesome!

a non-blocking suggestion but otherwise lgtm and huge improvement over the existing page

**Comment by Kyle-Verhoog** [comment]
@mergifyio backport 1.6

**Review by P403n1x87 — DISMISSED** [review]
😢  https://github.com/DataDog/envier/pull/7

File: docs/configuration.rst (hunk 1 above)
Comment (brettlangdon) [reply docs/configuration.rst:31]: ooooo! yes, that is a great idea!

**Comment by Kyle-Verhoog** [comment]
🔜 @P403n1x87 🔜

File: docs/configuration.rst (hunk 1 above)
Comment (brettlangdon) [reply docs/configuration.rst:31]:
```yaml
version_added:
  0.38.0: comma separated support added
  0.48.0: space separated support added
```

Then render at the bottom as:

```md
_Added in version v0.38.0_: comma separated support added

_Added in version v0.48.0_: space separated support added
```

?

**Comment by plannigan** [comment]
The new page layout looks great. Also appreciate that it is now possible to link directly to specific variables.

File: docs/conf.py [hunk 2]
```diff
@@ -673,7 +673,61 @@ def run(self):
...
+                        results.append("    *Changed in version {}*: {}".format(version, note), "", 0)
+                    else:
+                        results.append("    *New in version {}.*".format(version), "", 0)
+                    results.append("", "", 0)
+
+        # Generate the RST nodes to return for rendering
+        node = nodes.section()
+        node.document = self.state.document
+        self.state.nested_parse(results, 0, node)
+        # nested_parse_with_titles(self.state, results, node)
```
Comment (Kyle-Verhoog) [inline docs/conf.py:725]:
```suggestion
```

**Review by Kyle-Verhoog — DISMISSED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]

**Pull request merged**